### DISCUSSION TITLE
Is it possible to convert symbolic functions from matlab into a MX.sym function?

### DISCUSSION THREAD

**Discussion by trovlunde** [casadi/casadi#3290] [Q&A] [answered]
Hi,
I am running a program where the cost functions contains values of both MX.sym and sym types.

dist_fnc(x1,y1,x2,y2) = hypot(x1-x2, y1-y2);
error_sum = dist_fnc(x(1,:),x(2,:),bezi_x_values(x(7,:)), bezi_y_values(x(7,:)));
nmpfc_V = 0.001*sumsqr(u(1,:)) + 100*sumsqr(1-x(7,:)) + 1*sumsqr(error_sum);

The problem arises because x(1,:) and x(2,:) are MX.sym variables and the functions bezi_x/y_values is a classic symbolic matlab function. These functions are derived using the bernsteinMatrix function in Matlab and it does work with MX.sym. So I was wondering if there is an easy intuitive way to make a symbolic function into a MX.sym function. All my items have either arrived at error messages or being inable to evaluate the function afterwards.

In advance thanks for any replies!

**Comment by brunoml5** [comment, marked as answer]
Hi! Take a look at this #3172

**Comment by jgillis** [comment]
There is a conversion script to got from sym (mupad) expressions to CasADi expressions: https://gist.github.com/jgillis/58176f12c21680bc7e4fad4af9f33b2a
